portfolio/src/main/java/com/google/sps/servlets/LoadCommentServlet.java

```diff
@@ -37,29 +44,67 @@
 @WebServlet("/load-comments")
 public class LoadCommentServlet extends HttpServlet {
 
-  /** @private {!Array<{String user, String comment, String email, String userID, long id}>} */
+  /** @private {!Array<{String user, String comment, String email, String userID,float commentSentiment, long id}>} */
   private List<UserComments> commentArray = new ArrayList<>();
-  private int numberOfComments; // Number of displayed comments selected by the user.
+  private int numberOfComments; // Number of displayed comments selected by the user
+  private String languageCode; // Language abreviation following ISO 639 standard
   public static final String REDIRECT_URL = "/"; // Redirect to Portfolio
 
+  // Get the comment translation using Google's Translate API
+  public String translateComment(String comment) throws IOException {
+    Translate translateService = TranslateOptions.getDefaultInstance().getService();
+     // Do the comment translation.
+    Translation commentTranslation =
+        translateService.translate(comment, Translate.TranslateOption.targetLanguage(languageCode));
+    String translatedText = commentTranslation.getTranslatedText();
+
+    return translatedText;
+  }
+
+  /** Gets the translated comment in the requested language code */
+  public String getComment(String comment, Entity commentEntity, DatastoreService datastore) throws IOException {
+    EmbeddedEntity listOfTranslatedComments = 
+      (EmbeddedEntity) commentEntity.getProperty("comments");
+    String translatedCommentInLanguageCode = 
+      (String) listOfTranslatedComments.getProperty(languageCode);
+
+    // Translate the comment if the translation doesn't exist
+    if (translatedCommentInLanguageCode == null) {
+      String translatedComment = translateComment(comment);
+     
+      listOfTranslatedComments.setProperty(languageCode,translatedComment);
+      commentEntity.setProperty("comments", listOfTranslatedComments);
+      datastore.put(commentEntity);
+
+      return translatedComment;
+    }
+    return translatedCommentInLanguageCode;
```

Comment:
Nit: nothing wrong with using two return statements, but do you think you can find a way to use only one? For example:

```
if (translatedCommentInLanguageCode == null) {
    translatedCommentedInLanguageCode = translateComment(comment);
    // Do other things
}
return translatedCommentInLanguageCode;
```

Reply:
(I think you can reuse translatedCommentInLanguageCode)

Reply:
Sure can be done!